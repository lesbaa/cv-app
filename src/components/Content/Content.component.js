import React from 'react'
import Main from '~/components/Main'
import Title from '~/components/Title'
import Nav from '~/components/Nav'
import styles from './Content.styles'

const Content = ({
  blurbMarkup,
  title,
  nextSlide,
  previousSlide,
  palette,
}) => (
  <div
    className="Content"
  >
    <Title title={title} opacity={palette.offsetOpacity} />
    <Main>
      <span
        className="content-blurb"
        dangerouslySetInnerHTML={{ __html: blurbMarkup }} // eslint-disable-line
        opacity={palette.offsetOpacity}
      />
    </Main>
    <Nav
      nextSlide={nextSlide}
      previousSlide={previousSlide}
    />
    <style jsx>{styles}</style>
  </div>
)

// TODO prop types

export default Content
