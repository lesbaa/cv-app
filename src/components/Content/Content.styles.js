import css from 'styled-jsx/css'

export default css`
  .Content {
    display: inline-block;
    box-sizing: border-box;
    margin: 7.5vh;
    align-items: center;
    position: relative;
    font-size: 0.8em;
    user-select: none;
  }

  .content-blurb :global(.human-too) > :global(span) {
    margin-left: 0.5em;
  }

  .content-blurb :global(.human-too) > :global(span:last-of-type) {
    margin-left: 0em;
  }

  .content-blurb :global(.human-too) > :global(span:nth-of-type(1n)) {
    color: rgb(255, 50, 50);
  }

  .content-blurb :global(.human-too) > :global(span:nth-of-type(2n)) {
    color: rgb(60, 255, 60);
  }

  .content-blurb :global(.human-too) > :global(span:nth-of-type(3n)) {
    color: rgb(220, 100, 0);
  }

  .content-blurb :global(.human-too) > :global(span:nth-of-type(4n)) {
    color: rgb(0, 90, 240);
  }

  .content-blurb :global(.human-too) > :global(span:nth-of-type(5n)) {
    color: inherit;
  }

  @media screen and (min-width: 500px) {
    .Content {
      width: 30%;
    }
  }
`
