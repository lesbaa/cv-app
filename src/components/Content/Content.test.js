/* eslint-env jest */
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Content from './Content.component'

Enzyme.configure({
  adapter: new Adapter(),
})

describe('Content', () => {
  const wrapper = shallow(
    <Content
      store={{}}
      blurbMarkup=""
      title=""
      nextSlide=""
      previousSlide=""
      palette={{}}
      hasNav
    />
  )

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
