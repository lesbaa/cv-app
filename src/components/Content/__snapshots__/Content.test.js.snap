// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content matches its snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    blurbMarkup="<p>Hiya this is a paragraph from backend!<p>"
    hasNav={true}
    nextSlide="dev-skills"
    palette={
      Object {
        "colors": Array [
          "#eee6ee",
          "#d9d9d9",
        ],
        "gradient": "to bottom right, #eee6ee, #d9d9d9",
      }
    }
    previousSlide="hello"
    store={Object {}}
    title="Hiya this is a title!"
  >
    <div
      className="child-one"
    >
      Helooo!
    </div>
  </Content>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Title
          title="Hiya this is a title!"
        />,
        <Main>
          <span
            className="jsx-160578109 content-blurb"
          />
          <div
            className="child-one"
          >
            Helooo!
          </div>
        </Main>,
        <Connect(Nav)
          nextSlide="dev-skills"
          previousSlide="hello"
        />,
        <JSXStyle
          css=".Content.jsx-160578109{display:inline-block;box-sizing:border-box;margin:7.5vh;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;font-size:0.8em;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.content-blurb.jsx-160578109 .human-too>span{margin-left:0.5em;}.content-blurb.jsx-160578109 .human-too>span:last-of-type{margin-left:0em;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(1n){color:inherit;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(2n){color:#554;}@media screen and (min-width:500px){.Content.jsx-160578109{width:30%;}}"
          styleId="160578109"
        />,
      ],
      "className": "jsx-160578109 Content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "opacity": undefined,
          "title": "Hiya this is a title!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <span
              className="jsx-160578109 content-blurb"
            />,
            <div
              className="child-one"
            >
              Helooo!
            </div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "jsx-160578109 content-blurb",
              "opacity": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Helooo!",
              "className": "child-one",
            },
            "ref": null,
            "rendered": "Helooo!",
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "nextSlide": "dev-skills",
          "previousSlide": "hello",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".Content.jsx-160578109{display:inline-block;box-sizing:border-box;margin:7.5vh;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;font-size:0.8em;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.content-blurb.jsx-160578109 .human-too>span{margin-left:0.5em;}.content-blurb.jsx-160578109 .human-too>span:last-of-type{margin-left:0em;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(1n){color:inherit;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(2n){color:#554;}@media screen and (min-width:500px){.Content.jsx-160578109{width:30%;}}",
          "styleId": "160578109",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Title
            title="Hiya this is a title!"
          />,
          <Main>
            <span
              className="jsx-160578109 content-blurb"
            />
            <div
              className="child-one"
            >
              Helooo!
            </div>
          </Main>,
          <Connect(Nav)
            nextSlide="dev-skills"
            previousSlide="hello"
          />,
          <JSXStyle
            css=".Content.jsx-160578109{display:inline-block;box-sizing:border-box;margin:7.5vh;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;font-size:0.8em;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.content-blurb.jsx-160578109 .human-too>span{margin-left:0.5em;}.content-blurb.jsx-160578109 .human-too>span:last-of-type{margin-left:0em;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(1n){color:inherit;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(2n){color:#554;}@media screen and (min-width:500px){.Content.jsx-160578109{width:30%;}}"
            styleId="160578109"
          />,
        ],
        "className": "jsx-160578109 Content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "opacity": undefined,
            "title": "Hiya this is a title!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <span
                className="jsx-160578109 content-blurb"
              />,
              <div
                className="child-one"
              >
                Helooo!
              </div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "jsx-160578109 content-blurb",
                "opacity": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Helooo!",
                "className": "child-one",
              },
              "ref": null,
              "rendered": "Helooo!",
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "nextSlide": "dev-skills",
            "previousSlide": "hello",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".Content.jsx-160578109{display:inline-block;box-sizing:border-box;margin:7.5vh;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;position:relative;font-size:0.8em;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none;}.content-blurb.jsx-160578109 .human-too>span{margin-left:0.5em;}.content-blurb.jsx-160578109 .human-too>span:last-of-type{margin-left:0em;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(1n){color:inherit;}.content-blurb.jsx-160578109 .human-too>span:nth-of-type(2n){color:#554;}@media screen and (min-width:500px){.Content.jsx-160578109{width:30%;}}",
            "styleId": "160578109",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
